agent/verify_job.go

```diff
@@ -99,6 +116,19 @@ func (r *JobRunner) verifyJob(keySet jwk.Set) error {
 				return newInvalidSignatureError(fmt.Errorf("job %q was signed with signature %q, but the value of BUILDKITE_COMMAND (%q) does not match the value of step.command (%q)", r.conf.Job.ID, step.Signature.Value, jobCommand, step.Command))
 			}
 
+		case "env":
+			// Everything in the step env (post-matrix interpolation) must be
+			// present in the job env, and have equal values.
+			for name, stepEnvValue := range step.Env {
+				jobEnvValue, has := r.conf.Job.Env[name]
+				if !has {
+					return newInvalidSignatureError(fmt.Errorf("job %q was signed with signature %q, but step.env defines %s which is missing from the job environment", r.conf.Job.ID, step.Signature.Value, name))
+				}
+				if jobEnvValue != stepEnvValue {
+					return newInvalidSignatureError(fmt.Errorf("job %q was signed with signature %q, but the value of %s (%q) does not match the value of step.env[%s] (%q)", r.conf.Job.ID, step.Signature.Value, name, jobEnvValue, name, stepEnvValue))
```

Comment:
re the `job %q was signed with signature %q` prefix on both of these errors: do we need to output the value of the signature? i realise this is based on precedent that i started...

also, re: our chat with ashwin the other day, how much information do we want to/should we disclose when the signatures match, but something in the job has been messed with? would it be worth just saying something to the effect off "step signatures matched, but the job was different from what we expected"?

Reply:
How about I merge this, then we can follow up with a PR comprehensively changing these errors?

Reply:
a followup sounds like a good idea